packages/common/src/hooks/useFilterBy/useFilterBy.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from 'react';
+
+interface FilterByConditionByType {
+  string: 'equalTo' | 'like';
+  date: 'beforeOrEqualTo' | 'afterOrEqualTo' | 'equalTo';
```

Comment:
In general the types here are really hard ..

We need to send an object like:

```
{
comment: { equalTo: value },
otherPartyName: {like: value}
}
```
where there is an implicit AND for the fields